kvbc/include/db_adapter.h

```diff
@@ -4,5 +4,51 @@
 
 #pragma once
 
+#include "kv_types.hpp"
+#include <optional>
+#include <utility>
+
+namespace concord::kvbc {
+
+class NotFoundException : public std::runtime_error {
+ public:
+  NotFoundException(const std::string& error) : std::runtime_error(("NotFoundException: " + error).c_str()) {}
+  const char* what() const noexcept override { return std::runtime_error::what(); }
+};
+
+class IDbAdapter {
+ public:
+  // Returns the added block ID.
+  virtual BlockId addBlockToBlockchain(const SetOfKeyValuePairs& updates) = 0;
+
+  // Adds a block from its raw representation and a block ID.
+  // Typically called by state transfer when a block is received and needs to be added.
+  virtual void addRawBlock(const RawBlock& rawBlock, const BlockId& blockId) = 0;
+
+  // Get block in its raw form
+  virtual RawBlock getRawBlock(const BlockId& blockId) const = 0;
+
+  // If found, return the actual version and the value.
+  virtual std::optional<std::pair<Value, BlockId>> getValue(const Key&, const BlockId& blockVersion) const = 0;
+
+  // Delete a block from the database
+  virtual void deleteBlock(const BlockId& blockId) = 0;
+
+  // Used to retrieve the latest block.
+  virtual BlockId getLastBlockchainBlockId() const = 0;
+
+  /**
+   * Used to retrieve the last reachable block
+   * Searches for the last reachable block.
+   * From ST perspective, this is maximal block number N
+   * such that all blocks 1 <= i <= N exist.
+   * In the normal state, it should be equal to last block ID.
+   */
```

Comment:
I think this comment can be made clearer in terms of reachable and last. Also, when we start deleting blocks, genesis block will not always have ID = 1.

Reply:
yep, from the current beginning

Reply:
done